src/async_fuse/memfs/s3_metadata.rs

```diff
@@ -474,45 +470,41 @@ impl<S: S3BackEnd + Sync + Send + 'static, K: KVEngine + 'static> MetaData for S
             });
 
         let full_path = root_inode.full_path().to_owned();
-        meta.cache.write().await.insert(FUSE_ROOT_ID, root_inode);
-        meta.path2inum.write().await.insert(full_path, FUSE_ROOT_ID);
-
+        // insert two K/V pairs into KV engine
+        // 1. FUSE_ROOT_ID -> root_inode
+        meta.set_node_to_kv_engine(FUSE_ROOT_ID, root_inode).await;
+        // 2. full_path -> FUSE_ROOT_ID
+        meta.set_inum_to_kv_engine(&full_path, FUSE_ROOT_ID).await;
```

Comment:
"When discussing the concurrency problem in the current metadata implementation, after this pull request is merged, the infrastructure that the metadata is built upon will be completely changed. However, how the metadata will be refactored and provide concurrency safety will be addressed in future work. This will be addressed in subsequent pull requests.